### DISCUSSION TITLE
What are the required PHP extensions?

### DISCUSSION THREAD

**Discussion by IngoBelka** [datenstrom/community#609] [Ask a question] [answered]
Yellow requires some PHP modules, which aren't present in some standard installation of LAMP :

* common
* mbstring
* xmlrpc
* gd
* xml
* cli 
* curl
* zip

I had to install these on my Ubuntu server, to get Yellow run well.

Did I forgot some module?

**Comment by markseuffert** [comment, marked as answer]
At present you need the PHP extensions `curl gd mbstring zip`. [See troubleshooting](https://datenstrom.se/yellow/help/troubleshooting).

I thought that server requirements are automatically checked during installation. You should see a message like `Datenstrom Yellow requires PHP extension` with more information. This is done in the source code with function `checkRequirements()`. I am curious, can you tell which problems you encountered, what didn't work as expected, something we can do better?

**Reply by IngoBelka** [reply]
Mark, I had an older tutorial for Yellow to install. Now I want to write a documentation for me an my colleagues for further use. To document, what I did with Yellow, to get it in current state, is an unloved work, but it must been done. There where currently no troubles with Yellow.

**Reply by markseuffert** [reply]
Ah thanks for the clarification. I think we added troubleshooting for common problems last year, hopefully this can cut down your documentation work.  If you're done with your documentation and don't mind sharing, please send me a copy.

**Reply by IngoBelka** [reply]
I can't promise something, but if I get it ready, I'll send a copy. ;-)

**Comment by glow12121** [comment]
On Fedora 38, Yellow required `php-gd` and `php-zip` to start.
